### PR TITLE
Akka stream tracing

### PR DESCRIPTION
**Pull request by ptravers**
Commit adds a package that can be used to pass Spans through an Akka Stream.

As part of this the concept of a SpanContext had to be introduced this was taken from #69 

Some of this code was copied across from that PR.

### CODE REVIEW

File: project/MoneyBuild.scala [hunk 1]
```diff
@@ -285,6 +303,14 @@ object MoneyBuild extends Build {
     val codahaleVersion = "3.0.2"
     val apacheHttpClientVersion = "4.3.5"
 
+    val akkaV = "2.4.14"
```
Comment (pauljamescleary) [inline project/MoneyBuild.scala]: are we sure about these akka and akka-http versions?  Not sure if they are the current

File: money-akka/src/main/scala/com/comcast/money/akka/TracedFlow.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  def traceStageAndStop(key: String, stageLogic: In => Out, isSuccessful: Boolean = true): Unit = {
+    implicit val (inMessage, spanContext): (In, SpanContextWithStack) = grab[TracedIn](in)
+    tracer.startSpan(key)
+    val outMessage = stageLogic(inMessage)
+    push[TracedOut](out, (outMessage, spanContext))
+    spanContext.getAll.foreach(_ => tracer.stopSpan(isSuccessful))
+  }
+}
```
Comment (pauljamescleary) [inline money-akka/src/main/scala/com/comcast/money/akka/TracedFlow.scala]: needs a new line

**Comment by pauljamescleary** [comment]
before we merge this, we also need to build the license headers.

Also, please sign the CLA :)

File: project/MoneyBuild.scala (hunk 1 above)
Comment (ptravers) [reply project/MoneyBuild.scala]: Current would be `akkaV = 2.5.11` and `akka-http = 10.1.0` but we aren't running that as of right now on PAPI. Not sure what the effect would be of having money depend on the latest version. I can take a look.

File: money-akka/src/main/scala/com/comcast/money/akka/TracedFlow.scala (hunk 2 above)
Comment (ptravers) [reply money-akka/src/main/scala/com/comcast/money/akka/TracedFlow.scala]: Sure

**Comment by ptravers** [comment]
Hey yeah I would be happy to be the maintainer for it. That's not a problem.

**Pull request merged**